client/my-sites/themes/themes-selection.jsx

```diff
@@ -196,13 +199,20 @@ const ConnectedThemesSelection = connect(
 			number
 		};
 
+		const lastQuery = getLastThemeQuery( state, sourceSiteId ) || query;
+		const themesCount = getThemesFoundForQuery( state, sourceSiteId, lastQuery );
+
+		if ( ! isEqual( omit( query, PAGINATION_QUERY_KEYS ), omit( lastQuery, PAGINATION_QUERY_KEYS ) ) ) {
+			query.page = 1;
+		}
+
 		return {
 			query,
 			source: sourceSiteId,
 			siteSlug: getSiteSlug( state, siteId ),
-			themes: getThemesForQueryIgnoringPage( state, sourceSiteId, query ) || [],
-			themesCount: getThemesFoundForQuery( state, sourceSiteId, query ),
-			isRequesting: isRequestingThemesForQuery( state, sourceSiteId, query ),
+			themes: getThemesForQueryIgnoringPage( state, sourceSiteId, lastQuery ) || noThemes,
```

Comment:
Is there any need to change this to declare the `const noThemes` and use it here? It seems clearer to me if this just returns [].